Index: .idea/misc.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+><?xml version=\"1.0\" encoding=\"UTF-8\"?>\r\n<project version=\"4\">\r\n  <component name=\"ProjectRootManager\" version=\"2\" project-jdk-name=\"Python 3.7\" project-jdk-type=\"Python SDK\" />\r\n</project>
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/misc.xml b/.idea/misc.xml
--- a/.idea/misc.xml	
+++ b/.idea/misc.xml	
@@ -1,4 +1,4 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
-  <component name="ProjectRootManager" version="2" project-jdk-name="Python 3.7" project-jdk-type="Python SDK" />
+  <component name="ProjectRootManager" version="2" project-jdk-name="Python 3.9 (IoT)" project-jdk-type="Python SDK" />
 </project>
\ No newline at end of file
Index: Version2.0/main2.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Version2.0/main2.py b/Version2.0/main2.py
new file mode 100644
--- /dev/null	
+++ b/Version2.0/main2.py	
@@ -0,0 +1,30 @@
+import matplotlib.pyplot as plt
+import simpy
+from Network import Network
+from Node import Node
+
+
+env = simpy.Environment()
+no_of_nodes: int = 110
+network = Network(env, no_of_nodes)
+nodes = network.get_nodes()
+connections = network.get_connections()
+
+node = simpy.Resource(env, capacity=1)
+env.process(Node.source(self=nodes[0], env=env, number=5, interval=10))
+env.run()
+
+
+# PLOT NODES AND CONNECTIONS
+# for i in nodes:
+#     x = i.get_X()
+#     y = i.get_Y()
+#     plt.plot(x, y, 'ro')
+#     plt.annotate(i.get_rank(), (x, y))
+# for i in connections:
+#     from_x = i.get_node_from().get_X()
+#     from_y = i.get_node_from().get_Y()
+#     to_x = i.get_node_to().get_X()
+#     to_y = i.get_node_to().get_Y()
+#     plt.plot([from_x, to_x], [from_y, to_y], 'k')
+# plt.show()
Index: .idea/IoT.iml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+><?xml version=\"1.0\" encoding=\"UTF-8\"?>\r\n<module type=\"PYTHON_MODULE\" version=\"4\">\r\n  <component name=\"NewModuleRootManager\">\r\n    <content url=\"file://$MODULE_DIR$\">\r\n      <excludeFolder url=\"file://$MODULE_DIR$/venv\" />\r\n    </content>\r\n    <orderEntry type=\"jdk\" jdkName=\"Python 3.7\" jdkType=\"Python SDK\" />\r\n    <orderEntry type=\"sourceFolder\" forTests=\"false\" />\r\n  </component>\r\n  <component name=\"TestRunnerService\">\r\n    <option name=\"PROJECT_TEST_RUNNER\" value=\"Unittests\" />\r\n  </component>\r\n</module>
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/IoT.iml b/.idea/IoT.iml
--- a/.idea/IoT.iml	
+++ b/.idea/IoT.iml	
@@ -4,7 +4,7 @@
     <content url="file://$MODULE_DIR$">
       <excludeFolder url="file://$MODULE_DIR$/venv" />
     </content>
-    <orderEntry type="jdk" jdkName="Python 3.7" jdkType="Python SDK" />
+    <orderEntry type="inheritedJdk" />
     <orderEntry type="sourceFolder" forTests="false" />
   </component>
   <component name="TestRunnerService">
